FHEM/00_SIGNALduino.pm

```diff
@@ -450,6 +450,15 @@ SIGNALduino_flash($) {
 	delete($hash->{FLASH_RESULT}) if (exists($hash->{FLASH_RESULT}));
 
 	qx($hash->{helper}{avrdudecmd});
+
+	my $hardware=AttrVal($name,"hardware","");
+	if ($? != 0 && ($hardware eq "nano328" || $hardware eq "nanoCC1101"))
+	{
+		$hash->{logMethod}->($name ,3, "$name: ERROR: avrdude exited with error, try with 115200 baud");
+		$hash->{helper}{avrdudecmd} =~ s/57600/115200/g;	# new bootloader optiboot used 115200
```

Comment:
Die Herausforderung bei dieser Variante ist, dass im Test nicht abgeprüft werden kann, ob das Kommando einmal mit 57600 und danach mit 115200 Baud aufgerufen wurde, da in der Variable ja nur der letzte Eintrag steht.

Die Hardware ließe sich so womöglich universeller abfragen:
`$hardware=~ "^nano"`

Reply:
Diesen Trick kannte ich noch nicht. Ich habe jetzt probehalber erst mal fest eingetragen:

	if (AttrVal($name,"hardware","") =~ "^nano")
	{
		$hash->{logMethod}->($name ,3, "$name: try flash with transfer rates 57600 and 115200");
		$hash->{helper}{avrdudecmd} =  "avrdude -c arduino -b 57600 -P /dev/ttyUSB0 -p atmega328p -vv -U flash:w:./FHEM/firmware/SIGNALDuino_nano3283.3.1-exp.hex 2>./log/SIGNALduino-Flash.log || avrdude -c arduino -b 115200 -P /dev/ttyUSB0 -p atmega328p -vv -U flash:w:./FHEM/firmware/SIGNALDuino_nano3283.3.1-exp.hex 2>./log/SIGNALduino-Flash.log";
	}
	qx($hash->{helper}{avrdudecmd});

Getestet habe ich jetzt nur mit OptiBoot - da funktioniert das.

Reply:
Ich habe eine Idee für den code. Ich commite gleich mal was zum testen